SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -50,87 +52,110 @@ bool FSpatialGDKEditor::GenerateSchema(bool bFullScan)
 		}
 	}
 
-	bSchemaGeneratorRunning = true;
-
-	// 80/10/10 load assets / gen schema / garbage collection.
-	FScopedSlowTask Progress(100.f, LOCTEXT("GeneratingSchema", "Generating Schema..."));
-	Progress.MakeDialog(true);
-
-	RemoveEditorAssetLoadedCallback();
-
 	if (Schema::IsAssetReadOnly(SpatialConstants::SCHEMA_DATABASE_FILE_PATH))
 	{
-		bSchemaGeneratorRunning = false;
 		return false;
 	}
 
-	if (!Schema::LoadGeneratorStateFromSchemaDatabase(SpatialConstants::SCHEMA_DATABASE_FILE_PATH))
+	if (Method == FullAssetScan)
 	{
-		Schema::ResetSchemaGeneratorStateAndCleanupFolders();
-	}
+		// Make sure SchemaDatabase is not loaded.
```

Comment:
How come we do this?

Reply:
If it's loaded, an external process won't be able to write it because a file lock will be held.